geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/pubsub/AbstractSubCommandsIntegrationTest.java

```diff
@@ -0,0 +1,198 @@
+
+    mockSubscriber.punsubscribe();
+  }
+
+
+  @Test
+  public void channels_shouldReturnListOfMatchingChannels_withActiveChannelSubscribers_whenCalledWithPattern() {
+
+    List<String> expectedChannels = new ArrayList<>();
+    expectedChannels.add("foo");
+
+    Runnable runnable =
+        () -> subscriber.subscribe(mockSubscriber, "foo", "bar");
+
+    Thread subscriberThread = new Thread(runnable);
+    subscriberThread.start();
+    waitFor(() -> mockSubscriber.getSubscribedChannels() == 2);
+    List<String> result = introspector.pubsubChannels("fo*");
+
+    assertThat(result).containsExactlyInAnyOrderElementsOf(expectedChannels);
+  }
+
+  @Test
+  public void channels_should_returnEmptyArray_givenPatternWithNoMatches() {
+    List<String> result = introspector.pubsubChannels("fo*");
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void channels_shouldOnlyReturnChannelsWithActiveSubscribers() {
+    List<byte[]> expectedChannels = new ArrayList<>();
+    expectedChannels.add(Coder.stringToBytes("bar"));
+    Runnable runnable = () -> subscriber.subscribe(mockSubscriber, "foo", "bar");
+    Thread subscriberThread = new Thread(runnable);
+
+    subscriberThread.start();
+    waitFor(() -> mockSubscriber.getSubscribedChannels() == 2);
+    mockSubscriber.unsubscribe("foo");
+    waitFor(() -> mockSubscriber.getSubscribedChannels() == 1);
+    List<byte[]> result =
+        uncheckedCast(introspector.sendCommand(Protocol.Command.PUBSUB, PUBSUB_CHANNELS));
+
+    assertThat(result).containsExactlyInAnyOrderElementsOf(expectedChannels);
+  }
+
+  @Test
+  public void channels_shouldNotReturnDuplicates_givenMultipleSubscribersToSameChannel_whenCalledWithoutPattern() {
+
+    Jedis subscriber2 = new Jedis("localhost", getPort(), JEDIS_TIMEOUT);
+
+    MockSubscriber mockSubscriber2 = new MockSubscriber();
+    List<byte[]> expectedChannels = new ArrayList<>();
+    expectedChannels.add(Coder.stringToBytes("foo"));
+
+    Runnable runnable = () -> subscriber.subscribe(mockSubscriber, "foo");
+    Thread subscriber1Thread = new Thread(runnable);
+    subscriber1Thread.start();
+
+    Runnable runnable2 = () -> subscriber2.subscribe(mockSubscriber2, "foo");
+    Thread subscriber2Thread = new Thread(runnable2);
+    subscriber2Thread.start();
+
+    waitFor(() -> (mockSubscriber.getSubscribedChannels() == 1)
+        && (mockSubscriber2.getSubscribedChannels() == 1));
+
+    List<byte[]> result =
+        uncheckedCast(introspector
+            .sendCommand(Protocol.Command.PUBSUB, PUBSUB_CHANNELS));
+
+    assertThat(result).containsExactlyInAnyOrderElementsOf(expectedChannels);
+    assertThat(result.size()).isEqualTo(1);
+
+    mockSubscriber2.unsubscribe();
+    waitFor(() -> mockSubscriber2.getSubscribedChannels() == 0);
+
+    subscriber2.close();
+  }
+
+  private void waitFor(Callable<Boolean> booleanCallable) {
+    GeodeAwaitility.await()
+        .ignoreExceptions() // ignoring socket closed exceptions
```

Comment:
If it's only socket closed exceptions that you want to ignore here, it might be better to use `.ignoreExceptionsInstanceOf(SocketException.class)` rather than ignoring all exceptions.